Whisky/Views/Bottle/Pins/PinCreationView.swift

```diff
@@ -0,0 +1,116 @@
+    @State private var isValid: Bool = false
+    @State private var isDuplicate: Bool = false
+    @Environment(\.dismiss) private var dismiss
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text("pin.title")
+                .bold()
+            Divider()
+            HStack {
+                Text("pin.name")
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                TextField(String(), text: $newPinName)
+                    .frame(width: 180)
+                    .textFieldStyle(.roundedBorder)
+            }
+            HStack {
+                Text("pin.path")
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                Button("create.browse") {
+                    let panel = NSOpenPanel()
+                    panel.canChooseFiles = true
+                    panel.allowedContentTypes = [UTType.exe,
+                                                 UTType(exportedAs: "com.microsoft.msi-installer"),
+                                                 UTType(exportedAs: "com.microsoft.bat")]
+                    panel.directoryURL = newPinURL ?? bottle.url.appending(path: "drive_c")
+                    panel.canChooseDirectories = false
+                    panel.allowsMultipleSelection = false
+                    panel.canCreateDirectories = false
+                    panel.begin { result in
+                        guard result == .OK, let url = panel.urls.first else { return }
+                        let pattern = "\\.(exe|msi|bat)$"
+                        let oldDefaultName = (newPinURL ?? url).lastPathComponent
+                            .replacingOccurrences(of: pattern, with: "", options: .regularExpression)
+                        newPinURL = url
+                        // Only reset newPinName if the textbox hasn't been modified
+                        if newPinName.isEmpty || newPinName == oldDefaultName {
+                            newPinName = url.lastPathComponent
+                                .replacingOccurrences(of: pattern, with: "", options: .regularExpression)
+                        }
+                    }
+                }
+            }
+            if let newPinURL {
+                Text(newPinURL.prettyPath())
+                    .truncationMode(.middle)
+                    .lineLimit(2, reservesSpace: true)
+                    .help(newPinURL.prettyPath())
+            }
+            HStack {
+                Button("create.cancel") {
+                    dismiss()
+                }
+                .keyboardShortcut(.cancelAction)
+                .frame(maxWidth: .infinity, alignment: .trailing)
+                Button("pin.create") {
+                    guard let newPinURL else { return }
+
+                    let newlyCreatedPin = Program(name: newPinName, url: newPinURL, bottle: bottle)
+                    let existingProgram = bottle.programs.first(where: { program in
```

Comment:
`.contains` would be simpler

Reply:
`existingProgram` is used to test for both existence as well as `.pinned` in this block. If we switch to `.contains` here, we will still have to perform a similar lookup later whenever `.contains` is `true`. Granted, this should be far less often than _every time_, so the perceived performance impact would be less.